File: payment-service/src/main/java/com/hedvig/paymentservice/graphQl/types/BankAccount.java
package com.hedvig.paymentservice.graphQl.types;

import com.hedvig.paymentservice.query.member.entities.Member;
import lombok.Value;

@Value
public class BankAccount {

  String bankName;
  String descriptor;
  DirectDebitStatus directDebitStatus;

  public static BankAccount fromMember(Member m) {
    return new BankAccount(m.getBank(), m.getDescriptor(), fromMemberDirectStatus(m.getDirectDebitStatus()));
  }

  private static DirectDebitStatus fromMemberDirectStatus(com.hedvig.paymentservice.domain.payments.DirectDebitStatus s) {
    switch (s) {
      case CONNECTED:
        return DirectDebitStatus.ACTIVE;
      case PENDING:
        return DirectDebitStatus.PENDING;
      case DISCONNECTED:
        return DirectDebitStatus.NEEDS_SETUP;
      default:
        return null;
    }
  }
}

File: payment-service/src/main/java/com/hedvig/paymentservice/graphQl/types/BankAccount.kt
package com.hedvig.paymentservice.graphQl.types

import com.hedvig.paymentservice.query.directDebit.DirectDebitAccountOrder
import com.hedvig.paymentservice.query.member.entities.Member
import com.hedvig.paymentservice.domain.payments.DirectDebitStatus as DomainPaymentsDirectDebitStatus


data class BankAccount(
    val bankName: String,
    val descriptor: String,
    val directDebitStatus: DirectDebitStatus
) {
    companion object {
        private fun fromMemberDirectStatus(s: DomainPaymentsDirectDebitStatus): DirectDebitStatus {
            return when (s) {
                DomainPaymentsDirectDebitStatus.CONNECTED -> DirectDebitStatus.ACTIVE
                DomainPaymentsDirectDebitStatus.PENDING -> DirectDebitStatus.PENDING
                DomainPaymentsDirectDebitStatus.DISCONNECTED -> DirectDebitStatus.NEEDS_SETUP
            }
        }

        fun fromDirectDebitAccountOrder(order: DirectDebitAccountOrder): BankAccount {
            return BankAccount(
                order.bank!!,
                order.descriptor!!,
                fromMemberDirectStatus(order.directDebitStatus!!)
            )
        }
    }
}


